File: Haskell/exercism/bob/src/Bob.hs
module Bob (responseFor) where
import           Data.Char (isAlpha, isSpace, isUpper)

responseFor :: String -> String
responseFor xs
  | (all isUpper axs) && (not $ null axs) = "Whoa, chill out!"
  | null nxs = "Fine. Be that way!"
  | last nxs == '?' = "Sure."
  | otherwise = "Whatever."
  where
    nxs = filter (not . isSpace) xs
    axs = filter isAlpha nxs

File: Python/exerism/bob/bob.py
def hey(words: str) -> str:
    response = ("Whatever.", "Whoa, chill out!", "Sure.", "Fine. Be that way!")
    words = words.strip()
    if not words:
        return response[3]
    if words.isupper():
        return response[1]
    if words[-1] == '?':
        return response[2]
    return response[0]
